changelog/fragments/remove-new-cmd-for-helm.yaml

```diff
@@ -0,0 +1,17 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      The `operator-sdk new` command no longer supports scaffolding new Helm projects with the `--type=Helm` flag.
+      To scaffold new projects, users are expected to use `operator-sdk init --plugins=helm.operator-sdk.io/v1` as part of the
+      [new CLI]((https://sdk.operatorframework.io/docs/new-cli)) for Helm operators.
+
+      See `operator-sdk init --plugins=helm.operator-sdk.io/v1 -h` and the following doc on how to scaffold a new project
+      https://sdk.operatorframework.io/docs/helm/quickstart/. To migrate existing projects to the new layout see:
+      https://sdk.operatorframework.io/docs/helm/migration.
+
+    kind: "removal"
+
+    # Is this a breaking change?
+    breaking: false
```

Comment:
User-facing removals are _always_ breaking changes. The changelog generator should fail on this.

You'll need to add a migration section. On other PRs we've just been putting the PR title in the header and "TBD" in the body to keep these PRs moving and not to waste time on wordsmithing now.